sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -316,3 +317,111 @@ def test_pinjoint_pi():
     J = PinJoint('J', P, C, parent_axis=P.frame.x+P.frame.y+P.frame.z,
                 child_axis=-C.frame.x-C.frame.y-C.frame.z)
     assert J._generate_vector() == P.frame.x
+
+
+def test_slidingjoint():
+    P = Body('P')
+    C = Body('C')
+    x, v = dynamicsymbols('x_S, v_S')
+    S = PrismaticJoint('S', P, C)
+    assert S._name == 'S'
+    assert S.parent == P
+    assert S.child == C
+    assert S.coordinates == [x]
+    assert S.speeds == [v]
+    assert S.kdes == [v - x.diff(t)]
+    assert S.parent_axis == P.frame.x
+    assert S.child_axis == C.frame.x
+    assert S.child_point.pos_from(C.masscenter) == Vector(0)
+    assert S.parent_point.pos_from(P.masscenter) == Vector(0)
+    assert S.parent_point.pos_from(S.child_point) == - x * P.frame.x
+    assert P.masscenter.pos_from(C.masscenter) == - x * P.frame.x
+    assert C.masscenter.vel(P.frame) == v * P.frame.x
+    assert P.ang_vel_in(C) == 0
+
+    l, m = symbols('l m')
+    S = PrismaticJoint('S', P, C, parent_joint_pos= l * P.frame.x,
+                    child_joint_pos= m * C.frame.y, parent_axis = P.frame.z)
+
+    assert S.parent_axis == P.frame.z
+    assert S.child_point.pos_from(C.masscenter) == m * C.frame.y
+    assert S.parent_point.pos_from(P.masscenter) == l * P.frame.x
+    assert S.parent_point.pos_from(S.child_point) == - x * P.frame.z
+    assert P.masscenter.pos_from(C.masscenter) == - x * P.frame.x
```

Comment:
Should this have `l` and `m` in it?

Reply:
This should be `m*C.y - x*P.z + l*P.x`. Right?

Reply:
For future joint implementations you need to post a figure for these examples. Right now, I'm having to draw a figure and determine the solutions myself to check these results. If you post a figure I can look at it and the results and check what you've done without having to work out the whole problem myself.

Reply:
All the cases here are same as of `PinJoint`.

Reply:
Why isn't m and l in the position vector from mass center to mass center?

Reply:
> Why isn't m and l in the position vector from mass center to mass center?

Because `parent_joint_pos` is defined in `P.frame.x` and `child_joint_pos` is defined in `C.frame.y` but the `parent_axis` is `P.frame.z'. The algined axes are `P.z` and `C.x` , that's why `m` and 'l' are not in `pos_vector`.

If I do something like-
```
>>> S = PrismaticJoint('S', P, C, parent_joint_pos= l * P.frame.z, child_joint_pos= m * C.frame.x, parent_axis = P.frame.z)
>>> P.masscenter.pos_from(C.masscenter)
(-l - x_S(t))*P_frame.z + m*C_frame.x
```

This way we have `l` and `m` in `pos_vector` because in `PrismaticJoint` the `velocity` and `pos_vector` are always about `parent_axis`.

Reply:
I have added it to tests.

Reply:
I took these lines:

```python
l, m = symbols('l m')
S = PrismaticJoint('S', P, C, parent_joint_pos= l * P.frame.x, child_joint_pos= m * C.frame.y, parent_axis = P.frame.z)
```

and I drew this figure:

![image](https://user-images.githubusercontent.com/276007/124475704-4ced6980-dda2-11eb-8cae-cc807adc54b1.png)

The position vector (red r) from the mass center of the child to the parent is:

`m*C.y -x*C.x - l*P.x`

from what I can tell.

Please show my your figure and your vector calculation.

Reply:
C.y doesn't have to be -P.x (because this choice is arbitrary), but regardless of the direction m, l, and x should all be present.

Reply:
Yes, you are right, I got the problem , it is not in implementation but in writing tests , I'll fix it first thing tomorrow morning.

Reply:
Fixed!